### ISSUE TITLE
Why the source code is different from the published paper?

### ISSUE THREAD

**Issue by fjc523** [hailanyi/3D-Multi-Object-Tracker#5]
1. In your paper, three costs (appearance cost, motion cost and geometry cost) are used. But in the source code, it seems that only position differences are used.
`dis = (all_detections[...,0:3]-all_predictions[...,0:3])**2`
Does it mean that other costs are actullay not so important for tracking?

2. The prediction confidence is not updated as descripted in the paper (Equition 19), why？

So I wonder whether I can get the same performance decribed in README with the code. If not, can you make some time to updated the code？

Thank you for your good method and looking forword to your reply.

**Comment by Tsingularity** [comment]
@fjc523 could you please give the link to the paper you mentioned? thanks!

**Comment by fjc523** [comment]
> @fjc523 could you please give the link to the paper you mentioned? thanks!

https://ieeexplore.ieee.org/document/9352500

@Tsingularity I found this in http://www.cvlibs.net/datasets/kitti/eval_tracking_detail.php?result=4e01f46d142bdc04dc2d220caed50d0ec8ce29f9

**Comment by fjc523** [comment]
I also noticed that the code author will publish a better method on IJCAI 2021 which ranked first in KITTI tracking. 

> H. Wu, Q. Li, C. Wen, X. Li, X. Fan and C. Wang: Tracklet Proposal Network for Multi-Object Tracking on Point Clouds. IJCAI 2021.

But the conference will be held a month later. 

@hailanyi could you please give me a copy of the paper in advance or give the link of the project? 

Thank you!

**Comment by Tsingularity** [comment]
@fjc523 thanks!

**Comment by hailanyi** [comment]
@fjc523 1. yes, the other appearance cost, motion cost and geometry cost contribute less to the tracking performance. Especially after the KITTI updated the HOTA metrics. These tricks contribute only about 0.5% of HOTA compared with a dynamic distance-based association. To keep simplicity and support more detectors, this code did not use deep features and some not very useful association tricks. I also update a note in the readme. 
2. I have made some modification to adapt the HOTA metric.

The overall performance is same as detailed in readme. I recently also updated a new detector [CasA]( https://github.com/hailanyi/CasA). The detection results on the KITTI test set are also  released. One can use these detections to attain a similar performance as our results on the KITTI leader-board.
